src/shell/components/FieldTypeNumber.tsx

```diff
@@ -0,0 +1,94 @@
+import { useEffect, useRef } from "react";
+import { IconButton, Stack, TextField } from "@mui/material";
+import RemoveRoundedIcon from "@mui/icons-material/RemoveRounded";
+import AddRoundedIcon from "@mui/icons-material/AddRounded";
+
+import { NumberFormatInput } from "./NumberFormatInput";
+
+type FieldTypeNumberProps = {
+  required: boolean;
+  name: string;
+  value: string;
+  onChange: (value: string, name: string) => void;
+  hasError: boolean;
+};
+export const FieldTypeNumber = ({
+  required,
+  value,
+  onChange,
+  name,
+  hasError,
+}: FieldTypeNumberProps) => {
+  const numberInputRef = useRef(null);
+
+  useEffect(() => {
+    if (value === "0") {
+      numberInputRef.current?.setSelectionRange(1, 1);
+    }
+  }, [value]);
+
+  const modifyNumberValue = (action: "increment" | "decrement") => {
+    const integerFractionalSplit = value.toString().split(".");
+
+    switch (action) {
+      case "increment":
+        integerFractionalSplit[0] = (+integerFractionalSplit[0] + 1).toString();
+        break;
+
+      case "decrement":
+        integerFractionalSplit[0] = (+integerFractionalSplit[0] - 1).toString();
+        break;
+
+      default:
+        break;
+    }
+
+    onChange(integerFractionalSplit.join("."), name);
+  };
+
+  return (
+    <TextField
+      inputRef={numberInputRef}
+      variant="outlined"
+      fullWidth
+      value={value?.toString() || "0"}
+      name={name}
+      required={required}
+      onChange={(evt) => {
+        onChange(evt.target.value?.replace(/^0+/, "") ?? "0", name);
+      }}
+      onKeyDown={(evt) => {
```

Comment:
This actually resolves a different issue other than the default to 0 when empty. The cursor jumps before 0 instead of being after (which is how the input type number behaves) if I do that that's why I ended up just preventing the backspace and delete keys when the value is already 0.